classes/context/Context.inc.php

```diff
@@ -514,99 +515,36 @@ public function updateSetting($name, $value, $type = null, $isLocalized = false)
-        }
-
-        return $metricTypes;
-    }
-
-    /**
-    * Returns the currently configured default metric type for this context.
-    * If no specific metric type has been set for this context then the
-    * site-wide default metric type will be returned.
-    *
-    * @return null|string A metric type identifier or null if no default metric
-    *   type could be identified.
-    */
-    public function getDefaultMetricType()
+     * Whether to track usage statistics by institutions.
+     * Consider context setting only if the site setting is enabled and context setting disabled (= false).
+     */
+    public function isInstitutionStatsEnabled(Site $site): bool
     {
-        $defaultMetricType = $this->getData('defaultMetricType');
-
-        // Check whether the selected metric type is valid.
-        $availableMetrics = $this->getMetricTypes();
-        if (empty($defaultMetricType)) {
-            if (count($availableMetrics) === 1) {
-                // If there is only a single available metric then use it.
-                $defaultMetricType = $availableMetrics[0];
-            } else {
-                // Use the site-wide default metric.
-                $application = Application::get();
-                $defaultMetricType = $application->getDefaultMetricType();
-            }
-        } else {
-            if (!in_array($defaultMetricType, $availableMetrics)) {
-                return null;
-            }
+        $enableInstitutionUsageStats = $site->getData('enableInstitutionUsageStats');
+        if ($enableInstitutionUsageStats &&
+            ($this->getData('enableInstitutionUsageStats') !== null) && !$this->getData('enableInstitutionUsageStats')) {
+            $enableInstitutionUsageStats = $this->getData('enableInstitutionUsageStats');
         }
-
-        return $defaultMetricType;
-    }
-
-    /**
-    * Retrieve a statistics report pre-filtered on this context.
-    *
-    * @see <https://pkp.sfu.ca/wiki/index.php/OJSdeStatisticsConcept#Input_and_Output_Formats_.28Aggregation.2C_Filters.2C_Metrics_Data.29>
-    * for a full specification of the input and output format of this method.
-    *
-    * @param null|integer|array $metricType metrics selection
-    * @param int|array $columns column (aggregation level) selection
-    * @param array $orderBy order criteria
-    * @param null|DBResultRange $range paging specification
-    *
-    * @return null|array The selected data as a simple tabular
-    *  result set or null if metrics are not supported by this context.
-    */
-    public function getMetrics($metricType = null, $columns = [], $filter = [], $orderBy = [], $range = null)
-    {
-        // Add a context filter and run the report.
-        $filter[PKPStatisticsHelper::STATISTICS_DIMENSION_CONTEXT_ID] = $this->getId();
-        $application = Application::get();
-        return $application->getMetrics($metricType, $columns, $filter, $orderBy, $range);
+        return $enableInstitutionUsageStats;
```

Comment:
This will need a `(bool)` cast -- installations that upgraded from previous versions will not have this setting present, which will cause:
```
PHP Fatal error:  Uncaught TypeError: PKP\context\Context::isInstitutionStatsEnabled(): Return value must be of type bool, null returned in /home/asmecher/git/ojs-main/lib/pkp/classes/context/Context.inc.php:547
```

Reply:
(I suspect the other new boolean settings will be the same, in case you didn't catch them too!)